provider/agent.go

```diff
@@ -16,23 +16,31 @@ import (
 func agentResource() *schema.Resource {
 	return &schema.Resource{
 		Description: "Use this resource to associate an agent.",
-		CreateContext: func(c context.Context, resourceData *schema.ResourceData, i interface{}) diag.Diagnostics {
+		CreateContext: func(ctx context.Context, resourceData *schema.ResourceData, i interface{}) diag.Diagnostics {
```

Comment:
nit: I'm curious if there is a way to check if `CheckContext` called `updateStartupScriptBehaviorIfLoginBeforeReady` or if it was `ReadWithoutTimeout`.

BTW It would be nice to write a comment on why we need to call it in both contexts. I think that we usually stick with `ReadContext`.